bitcoin/BRPeer.c

```diff
@@ -1263,15 +1274,36 @@ void BRPeerDisconnect(BRPeer *peer)
 {
     BRPeerContext *ctx = (BRPeerContext *)peer;
     int socket = -1;
+    pthread_attr_t attr;
 
     if (_peerCheckAndGetSocket(ctx, &socket)) {
         pthread_mutex_lock(&ctx->lock);
         ctx->status = BRPeerStatusDisconnected;
         pthread_mutex_unlock(&ctx->lock);
 
         if (shutdown(socket, SHUT_RDWR) < 0) peer_log(peer, "%s", strerror(errno));
-        close(socket);
     }
+
```

Comment:
Likely the most controversial part; the spawning of a "cleanup"-like thread. Also, the dual nature of this function...

Reply:
```suggestion

```

Reply:
```suggestion

```